source/Meadow-Filter-Parse.js

```diff
@@ -100,6 +131,100 @@ const getFilterComparisonOperator = (pFilterOperator) =>
+ * @return {string} meadow-friendly type (e.g. SQL type that one could cast to)
+ */
+const getDataType = (encodedType) =>
+{
+	let type = 'CHAR';
+	switch (encodedType)
+	{
+		case 'SINT':
+			type = 'SIGNED';
+			break;
+		case 'UINT':
+			type = 'UNSIGNED';
+			break;
+		case 'DD':
+			// base 10, 5 decimal places (should cover 99% of use cases)
+			type = 'DECIMAL(10,5)';
+			break;
+		case 'DATE':
+			type = 'DATE';
+			break;
+		default:
+			break;
+	}
+	return type;
+};
+
+/**
+ * @spec fieldColumn is the target database (table) column; jsonPath is the target path to use for JSON_EXTRACT
+ * @param {object} pFilterStanza - a filter stanza with the properties defined in the interface
+ * @returns {object} structure { fieldColumn, jsonPath }
+ */
+const parseJSONFieldAndPath = (pFilterStanza) =>
+{
+	const splitIndex = pFilterStanza.Field.indexOf('.');
+	if (splitIndex < 1)
+	{
+		const msg = `Invalid format for Field[${pFilterStanza.Field}]`;
+		throw new Error(msg);
+	}
+	const fieldColumn = pFilterStanza.Field.substring(0, splitIndex);
+	const jsonPath = pFilterStanza.Field.substring(splitIndex);
+	const result =
+	{
+		fieldColumn,
+		jsonPath
+	};
+	return result;
+};
+
+/**
+ * @spec generates the appropriate filter(s) for looking inside a JSON field
+ * @param {string} fieldColumn - the target column for the meadow entity
+ * @param {string} jsonPath - the target chain (. notation) of possibly nested json keys in the value of 'fieldColumn'
+ * @param {number|string|Array|object} value - the value to use in the filter for comparison
+ * @param {string} comparisonOperator - the comparison operator (e.g. =, <=, NOT IN)
+ * @param {string} connector - the logical connector to this query (AND/OR)
+ * @param {object} pQuery - the foxhound query object
+ * @return {object} pQuery
+ */
+const addFilterJSONToQuery = (fieldColumn, jsonPath, value, comparisonOperator, connector, pQuery) =>
+{
+	pQuery.addFilter('', '', '(');
+	// NOTE: bypassing the table name injection by not providing a column argument
+	// TODO: prefix 'fieldColumn' with table name to avoid ambiguity (e.g. in joins)
+	pQuery.addFilter('', 1, `JSON_VALID(${fieldColumn}) =`, 'AND', 'validJson');
+	let command = `JSON_EXTRACT(${fieldColumn}, '$${jsonPath}')`;
+	let finalValue = value;
+	// NOTE: strings like "1 " are a number (1)
+	if (Array.isArray(value))
+	{
+		const nanFound = value.find((v) => isNaN(Number(v)));
+		if (!nanFound)
+		{
+			command = `JSON_UNQUOTE(${command})`;
+			finalValue = value.map((v) => Number(v));
+		}
+	}
+	else if (!isNaN(Number(value)))
+	{
+		// assumption: request contained a number embedded in a string, number comparison is desired
+		// value is a number, extract it from the strings
+		command = `JSON_UNQUOTE(${command})`;
+		finalValue = Number(value);
+	}
+	// NOTE: for whatever reason, foxhound accepts the JSON_UNQUOTE/JSON_EXTRACT commands without the bypass needed for JSON_VALID
```

Comment:
This seems sus. Would like to understand this better. What I observed when working on this with Anu was that the workaround was not needed if the field was fully qualified (had a `.` in it), which is why I am suspicious about that "qualifying" API code.